_dprhtml/js/bookmarks.js

```diff
@@ -1,39 +1,57 @@
-function bookmarkXML() {
-  var cont = readFile('DPR_Bookmarks');
-  cont = (cont ? cont.join('\n') : '<?xml version="1.0" encoding="UTF-8"?>\n<xml></xml>');
-  var parser=new DOMParser();
-  var xmlDoc = parser.parseFromString(cont,'text/xml');
-  return xmlDoc;
+function eraseBookmark(i) {
+  if (__navigationTabViewModel.isStorageSupportedByBrowser) {
+    let bookmarksArrayFromStorage = localStorage.getItem("bookmarksArray");
+    let data = [];
+    if (bookmarksArrayFromStorage) {
+      data = JSON.parse(bookmarksArrayFromStorage).slice();
+      data.splice(i, 1);
+      localStorage.setItem("bookmarksArray", JSON.stringify(data));
+      __navigationTabViewModel.updateBookmarks();
+      bookmarkframe(0);
+    }
+  }
 }
 
-function eraseBookmark(idx) {
-  var xmlDoc = bookmarkXML();
-  var thisNode = xmlDoc.getElementsByTagName('bookmark')[idx];
-  var answer = confirm('Are you sure you want to erase the bookmark "' + thisNode.getElementsByTagName('name')[0].textContent + '"?')
-  if(answer)
-  {
-        xmlDoc.documentElement.removeChild(thisNode);
+function clearBookmarks(gofrom) {
+  if (__navigationTabViewModel.isStorageSupportedByBrowser) {
+    var answer = confirm('Are you sure you want to erase all of the stored bookmarks?');
+    if(!answer) { return; }
+    let bookmarksArrayFromStorage = localStorage.getItem("bookmarksArray");
+    if (bookmarksArrayFromStorage) {
+      localStorage.removeItem("bookmarksArray");
+      __navigationTabViewModel.updateBookmarks();
+      bookmarkframe(0);
+    }
+  }
+}
 
-    var outfile = (new XMLSerializer()).serializeToString(xmlDoc);
-    writeFile('DPR_Bookmarks', outfile);
-    var sidebar = DPRSidebarWindow();
-    if (sidebar) {
-      sidebar.DPRNav.bookmarkBox();
+function getBookmarks() {
+  if (__navigationTabViewModel.isStorageSupportedByBrowser) {
+    let bookmarksArrayFromStorage = localStorage.getItem("bookmarksArray");
+    let content = [];
+    if (bookmarksArrayFromStorage) {
+      let data = JSON.parse(bookmarksArrayFromStorage);
+      for (var i in data) {
+        content.push(data[i]);
+      }
     }
-        bookmarkframe(1);
+    return content;
   }
 }
 
-function eraseBookmarks(gofrom) {
-  var answer = confirm('Are you sure you want to erase all of the stored bookmarks?')
-  if(answer)
-  {
-        eraseFile('DPR_Bookmarks');
-    var sidebar = DPRSidebarWindow();
-    if (sidebar) {
-      sidebar.DPRNav.bookmarkBox();
+function addBookmark() {
+  if (__navigationTabViewModel.isStorageSupportedByBrowser) {
+    let bookmarksArrayFromStorage = localStorage.getItem("bookmarksArray");
+    if (bookmarksArrayFromStorage) {
+      let value = `${$('#dialog-bookmarkInput').val()}@${__navigationTabViewModel.locPlace}`;
```

Comment:
there is ko binding already on dialog-bookmarkInput - lets use that and not jquery

in general - lets try to get away from hard coding IDs across the files - it makes the product brittle

Reply:
done